src/lib/mathlib/math/filter/AlphaFilter.hpp

```diff
@@ -89,6 +89,16 @@ class AlphaFilter
 		return true;
 	}
 
+	void setCutoffFreq(float cutoff_freq)
```

Comment:
This function could potentially be misleading in the case you call `setCutoffFreq(float cutoff_freq)` which sets `_time_constant` but then use `update(const T &sample)` for the update which assumes `setParameters(dt, _time_constant)` was already called. I don't have a great idea how to make it clear but see the potential for people to run into this 😬

Reply:
Yes, I was wondering about this too. Should we instead have two distinct classes? One where dt needs to be set during the `update` call and the other one that uses the `setParameters` to update the alpha.

Reply:
Seems overkill no? Would it be a performance hit to just always use the "One where dt needs to be set during the update" and in the cases it's static just pass a constant? Seems way easier compared to the multiple options for basically the same thing.

Reply:
It does seem like overkill, but having separate classes could make each individually a bit easier to use, harder to misuse. Having to know which update to call when used a certain way is a leaky abstraction.

Reply:
> know which update to call when used a certain way is a leaky abstraction

How about we always use the interface we pass dt to the update? How much value is there in having a statical dt update compared to always pass the same dt.